packages/admin/admin/src/common/buttons/feedback/FeedbackButton.tsx

```diff
@@ -0,0 +1,166 @@
+    tooltipSuccessMessage?: React.ReactNode;
+    tooltipErrorMessage?: React.ReactNode;
+}
+
+type FeedbackButtonDisplayState = "idle" | "loading" | "success" | "error";
+
+export function FeedbackButton(inProps: FeedbackButtonProps) {
+    const {
+        loading,
+        hasErrors,
+        children,
+        variant = "contained",
+        color = "primary",
+        classes,
+        disabled,
+        startIcon,
+        endIcon,
+        tooltipSuccessMessage,
+        tooltipErrorMessage,
+        slotProps,
+
+        ...restProps
+    } = useThemeProps({
+        props: inProps,
+        name: "CometAdminFeedbackButton",
+    });
+
+    const [displayState, setDisplayState] = React.useState<FeedbackButtonDisplayState>("idle");
+
+    const ownerState: OwnerState = {
+        displayState: displayState,
+    };
+
+    const resolveTooltipForDisplayState = (displayState: FeedbackButtonDisplayState) => {
+        if (displayState === "success") {
+            return "success";
+        } else if (displayState === "error") {
+            return "error";
+        } else {
+            return "neutral";
+        }
+    };
+
+    React.useEffect(() => {
+        let timeoutId: number | undefined;
+        let timeoutDuration: number | undefined;
+        let newDisplayState: FeedbackButtonDisplayState;
+
+        if (displayState === "idle" && loading) {
+            setDisplayState("loading");
+        } else if (displayState === "loading" && hasErrors) {
+            timeoutDuration = 500;
+            newDisplayState = "loading";
+        } else if (displayState === "loading" && !loading && !hasErrors) {
+            timeoutDuration = 500;
+            newDisplayState = "success";
+        } else if (displayState === "error") {
+            timeoutDuration = 5000;
+            newDisplayState = "idle";
+        } else if (displayState === "success") {
+            timeoutDuration = 2000;
+            newDisplayState = "idle";
+        }
+
+        if (timeoutDuration !== undefined) {
+            timeoutId = window.setTimeout(() => {
+                setDisplayState(newDisplayState);
+            }, timeoutDuration);
+        }
+        return () => {
+            if (timeoutId !== undefined) {
+                window.clearTimeout(timeoutId);
+            }
+        };
+    }, [displayState, loading, hasErrors]);
+
+    const tooltip = (
+        <Tooltip
+            title={displayState === "error" ? tooltipErrorMessage : tooltipSuccessMessage}
+            open={displayState === "error" || displayState === "success"}
+            placement={endIcon && !startIcon ? "top-end" : "top-start"}
+            variant={resolveTooltipForDisplayState(displayState)}
+            {...slotProps?.tooltip}
+        >
+            <span>{startIcon ? startIcon : endIcon}</span>
+        </Tooltip>
+    );
+
+    return (
+        <Root
+            ownerState={ownerState}
+            loading={loading}
+            {...slotProps}
```

Comment:
This should be the `slotProps` of the slot, not all of them. 

```suggestion
            {...slotProps?.root}
```